### DISCUSSION TITLE
Custom class attribute name.

### DISCUSSION THREAD

**Discussion by Giorno-Giovana** [tailwindlabs/tailwindcss#14009] [Ideas]
We use a custom react ui kit that uses the `UNSAFE_className ' attribute as props. Class compilation doesn't work for them. How about adding some way to customize your regular expression search?

```js
classNameVariants: ['UNSAFE_className']
```

```tsx
function App() {
  return <SomeCustomReactComponent UNSAFE_className={'mt-8'} />
}

function SomeCustomReactComponent({ UNSAFE_className }) {
   return <div className={UNSAFE_className} />
 }
```

**Comment by wongjn** [comment]
You could try customizing the [`tailwindCSS.classAttributes` setting](https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss#tailwindcss.classattributes):

> ### `tailwindCSS.classAttributes`
> The HTML attributes for which to provide class completions, hover previews, linting etc. Default: `class`, `className`, `ngClass`, `class:list`

**Reply by Giorno-Giovana** [reply]
You're talking about intellisense. I'm talking about the fact that tailwind don't generate classes, because the tailwind compiler doesn't perceive strings inside UNSAFE_className as something that needs to be compiled.

**Reply by wongjn** [reply]
I see! Thank you for the clarification.

Tailwind should generate classes inside `UNSAFE_className` as long as the `App()` function code is covered by the `content` file globs in the Tailwind configuration. This is because Tailwind is language-agnostic. It does not actually "run" your code in anyway. Tailwind scans for classes in files as plain text. You can think of it like Tailwind perceives every file as a `.txt` file instead of a `.js` file.

See [this Stackblitz project](https://stackblitz.com/edit/vitejs-vite-rn2mcq?file=src%2Fmain.jsx) of your code working.

**Reply by Giorno-Giovana** [reply]
Yes, I know what you're talking about. But in this case, we use a separate library, which we are not responsible for the code. And in the components provided by this library, classes are passed through `UNSAFE_className`, which is why type does not work. I suggest adding the capability to specify mapping attributes in `tailwind.tailwind.js`. For example:

```js
classNameVariants: ['UNSAFE_className']
```

Passing a list of html attributes to the array that should be perceived by tailwind as something that contains tailwind classes, and, therefore, should be taken by the tailwind compiler in the assembly

**Reply by wongjn** [reply]
> classes are passed through `UNSAFE_className`, which is why type does not work.

I'm not sure what you mean by "type" here.

> I suggest adding the capability to specify mapping attributes in `tailwind.tailwind.js`

`tailwind.tailwind.js` is not a file that Tailwind uses – do you perhaps mean `tailwind.config.js`?

> But in this case, we use a separate library, which we are not responsible for the code. And in the components provided by this library, classes are passed through `UNSAFE_className`

This shouldn't matter. As long as the the strings that you pass to `UNSAFE_className` are contained in the file that the `content` file globs cover, the Tailwind class names should be generated, as per [my previous comment](https://github.com/tailwindlabs/tailwindcss/discussions/14009#discussioncomment-10072739). Perhaps if you could provide an example project, it could help clarify matters?
